fish-rust/src/builtins/string/pad.rs

```diff
@@ -0,0 +1,104 @@
+use std::borrow::Cow;
+
+use super::{width_without_escapes, Arguments, Direction, StringError, StringSubCommand};
+use crate::builtins::shared::{io_streams_t, STATUS_CMD_OK};
+use crate::fallback::fish_wcwidth;
+use crate::ffi::parser_t;
+use crate::wchar::{wstr, WString, L};
+use crate::wgetopt::{wopt, woption, woption_argument_t::*};
+use crate::wutil::fish_wcstol;
+
+pub struct Pad {
+    char_to_pad: char,
+    pad_char_width: usize,
+    pad_from: Direction,
+    width: usize,
+}
+
+impl Default for Pad {
+    fn default() -> Self {
+        Self {
+            char_to_pad: ' ',
+            pad_char_width: 1,
+            pad_from: Direction::Left,
+            width: 0,
+        }
+    }
+}
+
+impl StringSubCommand<'_> for Pad {
+    const LONG_OPTIONS: &'static [woption<'static>] = &[
+        wopt(L!("quiet"), no_argument, 'q'),
+        wopt(L!("right"), no_argument, 'r'),
+        wopt(L!("width"), required_argument, 'w'),
```

Comment:
There is no `long_opt` with `--char` in `string.cpp`, it should be `--chars`, as `string pad -r -w 7 --chars - foo` works, and I just learned that we also allow shortening the options so e.g. `string pad -r -w 7 --ch - foo` works. This needs either a documentation fix or a potential breaking change

Reply:
#7341 - this is an unfortunate misbehavior of getopt (/getopt_long) that we inherited in our implementation.

We have thought of removing it, but it would be a breaking change that would have to be done with some care.